cmd/internal/server/handlers/fivetran_value_converters.go

```diff
@@ -214,3 +216,72 @@ func GetConverter(dataType fivetransdk.DataType) (ConverterFunc, error) {
 	}
 	return converter, nil
 }
+
+func GetEnumConverter(enumValues []string) (ConverterFunc, error) {
+	return func(value sqltypes.Value) (*fivetransdk.ValueType, error) {
+		parsedValue := value.ToString()
+		index, err := strconv.ParseInt(parsedValue, 10, 64)
```

Comment:
We should handle a string value of "NULL" here. as it stands right now, a value of `NULL` will fail the serialization, instead we should send a value type of null or the default value of the enum column as we do here : 
https://github.com/planetscale/fivetran-source/blob/05842c9db98fad6f9498e1a1303327d2f6104aed/cmd/internal/server/handlers/schema_aware_serializer.go#L112-L114

> If an ENUM column is declared to permit NULL, the NULL value is a valid value for the column, and the default value is NULL. If an ENUM column is declared NOT NULL, its default value is the first element of the list of permitted values.

Reply:
It seems like that conditional `if val.IsNull()` makes it so that the value is set to `fiventrasdk.ValueType_Null`, and the else case [where the enum serializer/set serializer would run](https://github.com/planetscale/fivetran-source/blob/05842c9db98fad6f9498e1a1303327d2f6104aed/cmd/internal/server/handlers/schema_aware_serializer.go#L116), never happens 🤔 is that not the case? 

Locally, it appears that null values are serialized correctly during the replication phase:
![Screenshot 2023-12-12 at 8 59 12 AM](https://github.com/planetscale/fivetran-source/assets/31225471/9c25e500-55a0-4d72-8c88-3d0590d62bb1)

Reply:
Unless, we're expecting a string that looks like "NULL"?

Reply:
I think `val.IsNull()` just short circuits this check and  might be all we need for ENUM types. 
The other thing is that this statement isn't true for PlanetScale databases 
> If an ENUM column is declared NOT NULL, its default value is the first element of the list of permitted values.

``` sql
show variables like "%sql_mode%";
+---------------+-----------------------------------------------------------------------------------------------------------------------+
| Variable_name | Value                                                                                                                 |
+---------------+-----------------------------------------------------------------------------------------------------------------------+
| sql_mode      | ONLY_FULL_GROUP_BY,STRICT_TRANS_TABLES,NO_ZERO_IN_DATE,NO_ZERO_DATE,ERROR_FOR_DIVISION_BY_ZERO,NO_ENGINE_SUBSTITUTION |
+---------------+-----------------------------------------------------------------------------------------------------------------------+
```

We set [STRICT_TRANS_TABLES](https://dev.mysql.com/doc/refman/8.0/en/sql-mode.html#sqlmode_strict_trans_tables) to true which will prevent a row from being created with an invalid ENUM value. 
TLDR: what you have is correct, we don't need a special override for NULL enum values.